### PR TITLE
Fix setuptools typing errors

### PR DESCRIPTION
**Pull request by Dreamsorcerer**

### CODE REVIEW

File: pytest.ini [hunk 1]
```diff
@@ -23,8 +23,8 @@ addopts =
   -p pytest_cov
 
   # `pytest-cov`:
-  --cov
-  --cov-context=test
+  --cov=yarl/
+  --cov=tests/
```
Comment (webknjaz) [inline pytest.ini:27]: No, this change is incorrect. The config must be in `.coveragerc` to enable it to be reused outside pytest.

File: .coveragerc [hunk 2]
```diff
@@ -12,10 +12,4 @@ show_missing = true
 
 [run]
 branch = true
-cover_pylib = false
 plugins = Cython.Coverage
-relative_files = true
-source =
-  tests
-source_pkgs =
-  yarl
```
Comment (webknjaz) [inline .coveragerc:21]: Keep this as is, it's configured intentionally. Are there any settings you didn't like?

**Review by webknjaz — CHANGES_REQUESTED** [review]
The config changes are incorrect. The typing might be fine, I didn't look into it. Note that the in-tree build-backend changes will need to be copied over into other projects, but I haven't yet copied the previous bugfix commit.

File: .coveragerc [hunk 3]
```diff
@@ -12,10 +12,4 @@ show_missing = true
 
 [run]
 branch = true
-cover_pylib = false
 plugins = Cython.Coverage
-relative_files = true
```
Comment (webknjaz) [inline .coveragerc:17]: Hm.. Perhaps this relative files setting could be a workaround for codecov not understanding paths… Could be an adequate repro for a bug report.

**Comment by commonism** [comment]
Can you merge the typing glue to fix the mypy/linter ci failure?

**Comment by webknjaz** [comment]
> Can you merge the typing glue to fix the mypy/linter ci failure?

Once it's re-submitted as a separate PR, I think so.

File: .coveragerc (hunk 2 above)
Comment (Dreamsorcerer) [reply .coveragerc]: Blindly copying from a repo which still works. I'll just merge the typing fixes and leave it to you.

**Pull request merged**